recipes/libjxl/all/conanfile.py

```diff
@@ -13,90 +16,110 @@ class LibjxlConan(ConanFile):
+        tc.variables["JPEGXL_ENABLE_BENCHMARK"] = False
+        tc.variables["JPEGXL_ENABLE_EXAMPLES"] = False
+        tc.variables["JPEGXL_ENABLE_MANPAGES"] = False
+        tc.variables["JPEGXL_ENABLE_SJPEG"] = False
+        tc.variables["JPEGXL_ENABLE_OPENEXR"] = False
+        tc.variables["JPEGXL_ENABLE_SKCMS"] = False
+        tc.variables["JPEGXL_ENABLE_TCMALLOC"] = False
+        tc.variables["JPEGXL_FORCE_SYSTEM_BROTLI"] = True
+        tc.variables["JPEGXL_FORCE_SYSTEM_HWY"] = True
+        tc.variables["JPEGXL_FORCE_SYSTEM_LCMS2"] = True
+        if cross_building(self):
+            tc.variables["CMAKE_SYSTEM_PROCESSOR"] = \
                 str(self.settings.arch)
-        self._cmake.configure()
-        return self._cmake
+        tc.variables["JPEGXL_ENABLE_TOOLS"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
 
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
         if self.options.shared:
             libs_dir = os.path.join(self.package_folder, "lib")
-            tools.remove_files_by_mask(libs_dir, "*.a")
-            tools.remove_files_by_mask(libs_dir, "*-static.lib")
-
-            if self.settings.os == "Windows":
-                self.copy("jxl_dec.dll", src="bin", dst="bin")
-                self.copy("jxl_dec.lib", src="lib", dst="lib")
-                for dll_path in glob.glob(os.path.join(libs_dir, "*.dll")):
-                    shutil.move(dll_path, os.path.join(self.package_folder,
-                                "bin", os.path.basename(dll_path)))
-            else:
-                self.copy("libjxl_dec.*", src="lib", dst="lib")
+            rm(self, pattern="*.a", folder=libs_dir)
+            rm(self, pattern="*-static.lib", folder=libs_dir)
 
     def _lib_name(self, name):
         if not self.options.shared and self.settings.os == "Windows":
             return name + "-static"
         return name
 
+    def _stdcpp_library(self):
+        libcxx = self.settings.get_safe("compiler.libcxx")
+        if libcxx in ["libstdc++", "libstdc++11"]:
+            return "stdc++"
+        elif libcxx in ["libc++"]:
+            return "c++"
+        elif libcxx in ["c++_shared"]:
+            return "c++_shared"
+        elif libcxx in ["c++_static"]:
+            return "c++_static"
+        return None
```

Comment:
there is stdcpp_library in conan.tools.build

Reply:
@SpaceIm 
Sorry for my late response.
I'm going to improve this PR with your comments.